metrics/processors/kubeprocessors/pod_based_enricher.go

```diff
@@ -196,7 +201,19 @@ func intValue(value int64) core.MetricValue {
 	}
 }
 
-func NewPodBasedEnricher(podLister *cache.StoreToPodLister) (*PodBasedEnricher, error) {
+func NewPodBasedEnricher(url *url.URL) (*PodBasedEnricher, error) {
```

Comment:
We pass pod listener for a reason. In large clusters with thousands of pods podsListeners are quite big (they contain a cache of ALL pods in the system with all metadata) and having 2 copies of them would be a big waste of memory.